### PR TITLE
Route list date scroll

### LINKED ISSUE #279: Scroll to previous/next day in route list

### PR DESCRIPTION
**Pull request by Kovler10**
Closes #279

### CODE REVIEW

File: app/screens/route-list/components/date-scroll.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  justifyContent: "space-evenly",
+}
+
+const TEXT_STYLE: TextStyle = {
+  width: "70%",
+  textAlign: "center",
+  color: color.primary,
+}
+
+type DateScrollDirection = "Forward" | "Backward"
```
Comment (planecore) [inline app/screens/route-list/components/date-scroll.tsx:54]: I think lowercased values are more appropriate here

File: app/screens/route-list/route-list-screen.tsx [hunk 2]
```diff
@@ -76,7 +86,21 @@ export const RouteListScreen = observer(function RouteListScreen({ navigation, r
...
+  const renderRouteCard = ({ item }: { item: RouteData }) => {
+    if (typeof item === "string") {
+      return (
+        <View
+          style={{
+            display: "flex",
+            alignItems: "center",
+            padding: "2%",
+            justifyContent: "center",
+          }}
```
Comment (planecore) [inline app/screens/route-list/route-list-screen.tsx:98]: Move to styles object at the top of the file. Also, don't use `%` in padding

File: app/screens/route-list/components/date-scroll.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      newDate.setDate(newDate.getDate() + value)
+
+      return newDate
+    },
+    [routePlan.date],
+  )
+
+  const setNewDate = useCallback(
+    (value: number) => {
+      routePlan.setDate(routePlan.date.setDate(routePlan.date.getDate() + value))
```
Comment (planecore) [inline app/screens/route-list/components/date-scroll.tsx:71]: Don't edit the user's date in the store. Generate a new state for the scroll requests

**Review by planecore — CHANGES_REQUESTED** [review]

**Comment by planecore** [comment]
Show loading indicator only when `routeData` is empty, and add a loading indicator to the date scroll component.

File: app/screens/route-list/route-list-screen.tsx (hunk 2 above)
Comment (Kovler10) [reply app/screens/route-list/route-list-screen.tsx]: 🤙🏻

File: app/screens/route-list/route-list-screen.tsx [hunk 4]
```diff
@@ -40,7 +41,11 @@ export const RouteListScreen = observer(function RouteListScreen({ navigation, r
 
   useEffect(() => {
     if (trains.isSuccess) {
-      setRouteData((prevData) => [...prevData, routePlan.date.toDateString(), ...trains.data])
+      setRouteData((prevData) =>
+        uniqBy([...prevData, routePlan.date.toDateString(), ...trains.data], (item) => {
+          return typeof item === "string" ? item : item.trains.map((train) => train.departureTimeString).join()
```
Comment (planecore) [inline app/screens/route-list/route-list-screen.tsx:46]: Unique by train numbers, not departure times as they can be the same

`item.trains.map((train) => train.trainNumber).join()`

**Pull request closed without merging**